app/scripts/utils/ndarray-assign.js

```diff
@@ -2,7 +2,7 @@ import cwise from 'cwise';
 
 const ndarrayAssign = (target, source) => {
   const numSource = +source;
-  const isScalar = !isNaN(numSource);
+  const isScalar = !isNaN(numSource); // eslint-disable-line no-restricted-globals
```

Comment:
@flekschas : The semantics of `Number.isNaN` are slightly different, and I'm reluctant to start introducing potential changes to the behavior. Do you feel confident about that change in this particular location, or would you like us to replace `isNaN` globally?

I would prefer to leave this PR as-is and come back later and look at the the `// eslint-disable-lines` which remain.

Reply:
Or do you think we should use `Number.isNaN(Number('1.2.3'))`, in order to get the same semantics? That just seems very verbose...

Reply:
@flekschas : Can you clarify what changes you want to see so that you're ok with this being merged?

My preferences, in order, are:
1. `isNaN(numSource); // eslint-disable-line no-restricted-global`
2. `Number.isNaN(Number(numSource))`
3. `Number.isNaN(numSource)`

The first two I could also apply globally, because they do not change behavior. The last introduces a behavior change: We could do that, but I would only want to introduce it slowly, rather than applying it everywhere at once.

Reply:
Sorry I am getting to many github notifications. Please ping me on Slack if I don't reply :)

**In short**: I suggest using `Number.isNaN(numSource)` (see below why)

**In long**: I honestly disagree with "leave this PR as-is and come back later". The whole point of linting (in my mind) is to fix issues and this is here is a good opportunity to make the code clearer.

I would simply convert the current statement to `Number.isNaN(numSource)` which is equivalent to `isNaN(numSource)` _in this case_. Note that `numSource = + source` is already converted to a number so it's either a number or not. If you feel like this is confusing `Number.isFinite(numSource)` would also work. (No negation of course).

Reply:
> Number.isNaN(numSource)

Can we just do this on a case by case basis only when we're sure that `numSource` is already a number? 

So on this particular line is fine. Others can be evaluated individually.

This is good that you're both bringing this up. I didn't know the semantic differences between `isNaN` and `Number.isNaN` before. In fact, I didn't even know that `Number.isNaN` existed.

Reply:
Thanks for pushing back on this. I see now why we can be confident about the behavior here, and I'll give some more time when I look at other cases.